src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -553,4 +551,81 @@ public void refreshStudyInformation() {
+                new SearchTerm( participantIdPropertyTypeId.toString() + ":\"" + participantId + "\"", 0, 1 )
+        );
+        if ( result.getHits().size() != 1 ) {
+            return null;
+        }
+        SetMultimap<FullQualifiedName, Object> data = result.getHits().iterator().next();
+        if ( data.containsKey( INTERNAL_ID_FQN ) ) {
+            return UUID.fromString( data.get( INTERNAL_ID_FQN ).iterator().next().toString() );
+        }
+        return null;
+    }
+
+    @Override
+    public ParticipationStatus getParticipationStatus( UUID studyId, String participantId ) {
+
+        EdmApi edmApi;
+        SearchApi searchApi;
+
+        try {
+            ApiClient apiClient = apiClientCache.get( ApiClient.class );
+            edmApi = apiClient.getEdmApi();
+            searchApi = apiClient.getSearchApi();
+        } catch ( ExecutionException e ) {
+            logger.error( "Unable to load apis." );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        String participantsEntitySetName = ChronicleServerUtil.getParticipantEntitySetName( studyId );
+        UUID participantsEntitySetId = edmApi.getEntitySetId( participantsEntitySetName );
+        if ( participantsEntitySetId == null ) {
+            logger.error( "unable to get the participants EntitySet id for studyId = {}", studyId );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        UUID participantEntityKeyId = getParticipantEntityKeyId( participantId, participantsEntitySetId, searchApi );
+        if ( participantEntityKeyId == null ) {
+            logger.error( "unable to load participant entity key id for participantId = {}", participantId );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        Map<UUID, List<NeighborEntityDetails>> neighborResults = searchApi.executeFilteredEntityNeighborSearch(
+                participantsEntitySetId,
+                new EntityNeighborsFilter(
+                        Sets.newHashSet( participantEntityKeyId ),
+                        java.util.Optional.of( Sets.newHashSet( participantsEntitySetId ) ),
+                        java.util.Optional.of( Sets.newHashSet( studyEntitySetId ) ),
+                        java.util.Optional.of( Sets.newHashSet( participatedInEntitySetId ) )
+                )
+        );
+
+        java.util.Optional<SetMultimap<FullQualifiedName, Object>> targetAssociationDetails = neighborResults
+                .get( participantEntityKeyId )
+                .stream()
+                .filter( neighborResult -> neighborResult.getNeighborDetails().isPresent() )
+                .filter( neighborResult -> neighborResult
+                       .getNeighborDetails()
+                       .get()
+                       .get( STRING_ID_FQN )
+                       .equals( Sets.newHashSet( studyId ) )
+                )
+                .map( NeighborEntityDetails::getAssociationDetails )
+                .findFirst();
```

Comment:
is the status guaranteed to be the same on all associations?

Reply:
No. At the time of this implementation, no association entity has the status; the property was recently added.

Reply:
Actually, I'm not quite sure what you mean. Can you expand further on your question?

Reply:
there could theoretically be several (associationEntity, neighborEntity) pairs in the neighbors response and this is only checking the first. Are all associationEntities guaranteed to have the same status value here? Or is it guaranteed that there will only be one (associationEntity, neighborEntity) pair?

Reply:
I don't think anything is guaranteed. We expect there to only be one edge between the participant and the study, and `.filter()` should reduce to only one result where `"general.stringid" == studyId`.

Not sure how to handle the theoretically possible case where there's more than one edge between the participant and the study. Suggestions?

Reply:
hmm not sure without collecting to a collection